src/webgl/loading.js

```diff
@@ -158,6 +158,57 @@ p5.prototype.loadModel = function(path,options) {
   model.gid = `${path}|${normalize}`;
   const self = this;
 
+  async function getMaterials(lines){
+    const parsedMaterialPromises=[];
+    const mtlPaths=[];
+    return new Promise(async (resolve,reject)=>{
+      for (let i = 0; i < lines.length; i++) {
+        const mtllibMatch = lines[i].match(/^mtllib (.+)/);
+        if (mtllibMatch) {
+          let mtlPath='';
+          const mtlFilename = mtllibMatch[1];
+          const objPathParts = path.split('/');
+          if(objPathParts.length > 1){
+            objPathParts.pop();
+            const objFolderPath = objPathParts.join('/');
+            mtlPath = objFolderPath + '/' + mtlFilename;
+          }else{
+            mtlPath = mtlFilename;
+          }
+          try {
+            if(await fileExists(mtlPath)){
```

Comment:
This `await` will also cause us to do these `HEAD` requests one at a time instead of parallel. Maybe we can do something like:

```js
parsedMaterialPromises.push(
  fileExists(mtlPath).then(exists => {
    if (exists) {
      return parseMtl(self, mtlPath);
    } else {
      console.warn('MTL file not found or error in parsing; proceeding without materials.');
      return {};
    }
  }).catch(error => {
    console.warn('Error loading MTL file:', error);
    return {};
  })
);
```
(Also, while we're at it, maybe we can add the URL for the mtl file that we couldn't find to the warning we log?)